portal-web/test/functional/com/liferay/portalweb/tests/enduseree/formsworkflow/metrics/WorkflowMetrics.testcase

```diff
@@ -380,6 +380,135 @@ definition {
 			taskCount = "1");
 	}
 
+	@description = "LRQA-53662 Automate LPS-92053 View process item detail page"
+	@priority = "5"
+	test ViewProcessItemDetailPage {
+		property portal.acceptance = "true";
+
+		ProductMenu.gotoPortlet(
+			category = "Configuration",
+			panel = "Site Administration",
+			portlet = "Workflow");
+
+		Workflow.configureWorkflow(
+			workflowDefinition = "Single Approver",
+			workflowResourceValue = "Blogs Entry");
+
+		JSONBlog.addEntry(
+			entryContent = "Blogs Entry1 Content",
+			entryTitle = "Blogs Entry Title");
+
+		WorkflowMetrics.goToWorkflowProcessMetrics(workflowProcessName = "Single Approver");
+
+		WorkflowMetrics.goToDashboardAllItemsWithFilter(
+			workflowFilterOption = "PENDING",
+			workflowMetricsTab = "DASHBOARD",
+			workflowProcess = "Single Approver");
+
+		AssertElementPresent(locator1 = "WorkflowAllItems#WORKFLOW_INSTANCE_ID");
+
+		var instanceId = selenium.getText("//td[contains(@class,'table-title')]/a[@data-testid='instanceIdLink']/strong");
+
+		echo(" The instanceId is: ${instanceId}");
+
+		Click(locator1 = "WorkflowAllItems#WORKFLOW_INSTANCE_ID");
+
+		AssertTextEquals(
+			locator1 = "WorkflowAllItems#DETAIL_PAGE_TITLE",
+			value1 = "Item #${instanceId}");
+
+		AssertTextEquals(
+			locator1 = "WorkflowAllItems#DETAIL_PAGE_SLA_MESSAGE",
+			value1 = "No SLA records for this item.");
+
+		AssertTextEquals(
+			locator1 = "WorkflowAllItems#DETAIL_PAGE_PROCESS_STATUS",
+			value1 = "Pending");
+
+		AssertTextEquals(
+			locator1 = "WorkflowAllItems#DETAIL_PAGE_CREATED_BY",
+			value1 = "Test Test");
+
+		var currentDateDay = DateUtil.getCurrentDate();
+		var currentDateMonth = DateUtil.getCurrentMonthName();
+		var currentDateMonthShort = StringUtil.substring("${currentDateMonth}", "0", "3");
+
+		echo(" The current Month is: "${currentDateMonthShort}" ");
+
+		var currentDateYear = DateUtil.getCurrentYear();
+		var currentDate = "${currentDateMonthShort} ${currentDateDay}, ${currentDateYear}";
+
+		AssertTextEquals.assertPartialText(
+			locator1 = "WorkflowAllItems#DETAIL_PAGE_CREATION_DATE",
+			value1 = "${currentDate}");
+
+		AssertTextEquals(
+			locator1 = "WorkflowAllItems#DETAIL_PAGE_ASSET_TYPE",
+			value1 = "Blogs Entry");
+
+		AssertTextEquals(
+			locator1 = "WorkflowAllItems#DETAIL_PAGE_ASSET_TITLE",
+			value1 = "Blogs Entry Title");
+
+		AssertTextEquals(
+			locator1 = "WorkflowAllItems#DETAIL_PAGE_CURRENT_STEP",
+			value1 = "Review");
+
+		AssertTextEquals(
+			locator1 = "WorkflowAllItems#DETAIL_PAGE_CURRENT_ASSIGNEE",
+			value1 = "Unassigned");
+
+		Click(locator1 = "WorkflowAllItems#DETAIL_PAGE_CLOSE");
```

Comment:
Hi @melodywu ,
You indeed have got the objective of the test, good job! The test needs to validate exactly these fields you validated.
Though, you can notice that the way you implemented, if we wanted to use this validation again (as we would in the next lines when the instance is completed) we would have to copy the same lines again. Also, here it is working only for assignee: Test Test, Blogs instance, value: Pending, step: Review, etc...

I suggest you to create a macro for these validations. A macro like, I don't know, "validateInstanceDetails". This macro would receive all the parameters that should be validated:

${assigneeName}
${instanceID}
${CurrentStep}
${Creator}
${assetType}
etc...
(creationDate I believe that it could be calculated inside the macro, no need to pass a parameter)
Creating it that way would give us more flexibility to validate details from any type of asset, passing the parameters we want and avoiding duplicated code.

Please let me know if you have any questions about it